geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterLayer.scala

```diff
@@ -93,6 +93,45 @@ abstract class TiledRasterLayer[K: SpatialComponent: JsonFormat: ClassTag: Bound
     mask(geometries)
   }
 
+  def mask(
+    wkbRDD: RDD[Array[Byte]],
+    partitionStrategy: PartitionStrategy,
+    options: Rasterizer.Options
+  ): TiledRasterLayer[K] = {
+    val geometryRDD: RDD[Geometry] = wkbRDD.map { WKB.read }
+    val clipped: RDD[(SpatialKey, Geometry)] = geometryRDD.clipToGrid(rdd.metadata.layout)
+
+    val addValue = (arr: ArrayBuffer[Geometry], g: Geometry) => g +=: arr
+    val combineArrs = (arr1: ArrayBuffer[Geometry], arr2: ArrayBuffer[Geometry]) =>
+      if (arr1.size > arr2.size) arr2 ++=: arr1 else arr1 ++=: arr2
+
+    val aggregatedRDD: RDD[(SpatialKey, Seq[Geometry])] =
+      rdd.partitioner match {
+        case Some(p) =>
+          clipped.aggregateByKey(ArrayBuffer[Geometry](), p)(addValue, combineArrs).mapValues { _.toSeq }
+        case None =>
+          clipped.aggregateByKey(ArrayBuffer[Geometry]())(addValue, combineArrs).mapValues { _.toSeq }
+      }
+
+    /*
+     * The reason why this produces a () => Option[Partitioner] instead of just an Option[Partitioner]
+     * is due to type erasure when calling this methods from Python.
+     */
```

Comment:
Could you explain that problem this avoids? I would think that `() => Option[Partitioner]` would become `Function1[Any, Any]` when provided through `py4j` and would face the same problem as `Option[Partitioner]`.

Reply:
That's apparently not the case. When `partitionStrategy` is given as a `None` from Python it will be turned into a `null` that can be interpreted as either a `PartitionStrategy` or as an `Option`. What was happening was that instead of going `TiledRasterLayer.mask` -> `SpatialTiledRasterLayer.mask` it was going right to the `SpatialTiledRasterLayer.mask` and getting a match error on `null`.

Reply:
Right so it wouldn't be able to distinguish between:

```scala
def mask(
    aggregatedRDD: RDD[(SpatialKey, Seq[Geometry])],
    partitionStrategy: PartitionStrategy,
    options: Rasterizer.Options
  ): TiledRasterLayer[K]

// and

  def mask(
    wkbRDD: RDD[Array[Byte]],
    partitionStrategy: PartitionStrategy,
    options: Rasterizer.Options
  ): TiledRasterLayer[K] = {
```
because after erasure `RDD[Array[Byte]]` and `RDD[(SpatialKey, Seq[Geometry])]` look exactly the same: `RDD[Any]`. I guess it just picks the one it found first during reflection. So changing  `PartitionStrategy` to `() => Option[Partitioner]` allows it to distinguish between the two as its difference between the two.

I would say that its better to rename one of the `mask` methods rather than introducing a another way to specify the partitioner to avoid this problem.